runtime/lua/vim/loader.lua

```diff
@@ -0,0 +1,528 @@
+local uv = vim.loop
+
+local M = {}
+
+---@alias CacheHash {mtime: {sec:number, nsec:number}, size:number}
+---@alias CacheEntry {hash:CacheHash, chunk:string}
+
+---@class ModuleFindOpts
+---@field all? boolean Search for all matches (defaults to `false`)
+---@field rtp? boolean Search for modname in the runtime path (defaults to `true`)
+---@field patterns? string[] Paterns to use (defaults to `{"/init.lua", ".lua"}`)
+---@field paths? string[] Extra paths to search for modname
+
+---@class ModuleInfo
+---@field modpath string Path of the module
+---@field modname string Name of the module
+---@field stat? uv_fs_t File stat of the module path
+
+M.VERSION = 3
+M.path = vim.fn.stdpath('cache') .. '/luac'
+M.enabled = false
+---@type table<string, {total:number, time:number, [string]:number?}?>
+M.stats = {
+  find = { total = 0, time = 0, not_found = 0 },
+}
+
+---@class Loader
+---@field _rtp string[]
+---@field _rtp_pure string[]
+---@field _rtp_key string
+local Loader = {
+  ---@type table<string, table<string,ModuleInfo>>
+  _indexed = {},
+  ---@type table<string, string[]>
+  _topmods = {},
+  _loadfile = loadfile,
+}
+
+--- Tracks the time spent in a function
+---@private
+function M._track(stat, start)
+  M.stats[stat] = M.stats[stat] or { total = 0, time = 0 }
```

Comment:
should `M.stats`  be local to the module rather than on `M` itself?

Reply:
I currently show those stats in the Lazy debug ui, but that's not that important.

I'll localize `stats` and will for now add it to `M._inspect()`, which is already private. If that method would be removed at some point, then no big deal. I'll then just remove the debug output from the lazy ui.

Reply:
Made stats local and changed the private `M._inspect` method to return the raw stats and optionally print them